src/app/views/dashboard/student-dashboard/student-dashboard.component.html

```diff
@@ -1 +1,69 @@
-<p>student-dashboard works!</p>
+<div class="student-dash-main">
+  <h2 style="width: 100%;">Student DashBoard</h2>
+  <mat-card class="student-dash-card">
+    <mat-card-header>
+      <mat-card-title>Total Test Taken</mat-card-title>
+    </mat-card-header>
+    <mat-card-content>
+      <ngx-charts-advanced-pie-chart [view]="[400, 200]" [scheme]="colorScheme" [results]="resultStats"
+        [gradient]="gradient">
+      </ngx-charts-advanced-pie-chart>
+    </mat-card-content>
+  </mat-card>
+
+  <div class="student-dash-list-grid">
+    <div class="student-result-container mat-elevation-z8">
+      <!-- <div class="loading-shade" *ngIf="isLoadingResults || isRateLimitReached">
+        <mat-spinner *ngIf="isLoadingResults"></mat-spinner>
+        <div class="error-occured" *ngIf="isRateLimitReached">
+          Something is wrong with server.
+        </div>
+      </div> -->
+
+      <div class="example-table-container">
+
+        <table mat-table [dataSource]="dataSource" class="example-table">
+
+          <ng-container matColumnDef="name">
+            <th mat-header-cell *matHeaderCellDef>Test Name</th>
+            <td mat-cell *matCellDef="let row">{{row.name}}</td>
+          </ng-container>
+
+          <ng-container matColumnDef="totalScore">
+            <th mat-header-cell *matHeaderCellDef>Total Score</th>
+            <td mat-cell *matCellDef="let row">{{row.totalScore}}</td>
+          </ng-container>
+
+          <ng-container matColumnDef="markObtained">
+            <th mat-header-cell *matHeaderCellDef >Mark Obtained</th>
+            <td mat-cell *matCellDef="let row">{{row.markObtained}}</td>
+          </ng-container>
+
+          <ng-container matColumnDef="percentage">
+            <th mat-header-cell *matHeaderCellDef>Percentage</th>
+            <td mat-cell *matCellDef="let row">{{row.percentage}}%</td>
+          </ng-container>
+
+          <ng-container matColumnDef="status">
+            <th mat-header-cell *matHeaderCellDef>Status</th>
+            <td mat-cell *matCellDef="let row" [style.color]="row.color">{{row.status}}</td>
+          </ng-container>
+
+           <!-- actions Column -->
+           <ng-container matColumnDef="actions">
+            <th mat-header-cell *matHeaderCellDef>Actions</th>
+            <td mat-cell *matCellDef="let row">
+              <span class="action-btn" matTooltip="View Result" (click)="viewResult(row)"><mat-icon>view_compact</mat-icon></span>
+            </td>
+          </ng-container>
+
+          <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+          <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+        </table>
+      </div>
+
+      <mat-paginator [pageSizeOptions]="[2,5, 10, 25, 100]"></mat-paginator>
```

Comment:
make this page size option as a common attribute in a modal and read it everywhere.

Reply:
Changes in upcomming pr